src/v/schema/registry.cc

```diff
@@ -126,23 +126,19 @@ registry::get_valid_schema(ppsr::schema_id schema_id) const {
             co_return std::nullopt;
         }
     }
-    auto& schema_def = *schema_def_opt;
-    auto schema_type = schema_def.type();
-    switch (schema_type) {
+    auto schema_def = schema_def_opt->share();
+    switch (schema_def.type()) {
```

Comment:
Just want to confirm that a shared iobuf is what we want here, since in the previous regime we would move directly out of the `schema_definition` returned from `maybe_get`.

Looks like that one is shared from the schema store anyway, but the totality of ownership isn't immediately clear to me.

Reply:
🤦 ah yea no good reason